### DISCUSSION TITLE
filtering multiple choice fields

### DISCUSSION THREAD

**Discussion by z89** [pocketbase/pocketbase#434] [Q&A] [answered]
Hi, so I have a multiple choice field in my records in the form of a category . For example, a single record could have 
"category": ["1"] 
or even 
"category": ["1", "2"]

My issue is I can't seem to filter the records based off if they match a certain category? In other words, how do I filter a multiple choice field in a URL fetch? I can currently filter string fields based off a string like so:
http://localhost:8090/api/collections/products/records?page=1&perPage=50&filter=name="example"

However I am struggling to filter a multiple choice since instead of a string, it's an array of strings, which should look like so:
http://localhost:8090/api/collections/products/records?page=1&perPage=50&filter=category=["1","2"]

But that doesn't seem to work even with URL encoding. I've been stuck on this all day and still can't figure it out, and also can't work it out from the documentation.

Any help is appreciated, thanks!

**Comment by ganigeorgiev** [comment, marked as answer]
Currently there is no `IN` operator support via the filter syntax.
But if your category options are unique, you can apply a "like/contains" operation, eg. `?filter=category ~ '1' || category ~ '2'`.

_Since a simple "IN" filter query was requested previously, I'll consider implementing it for multy-select fields in a future release, but I'll think a little more on it after the ongoing users/profiles refactoring in #376._

**Reply by z89** [reply]
Thanks for the quick response.

It would be very useful to have the `IN` operator, but I will work out a solution since it's not implemented yet. Keep up the good work 👍🏼

**Reply by shubhendumadhukar** [reply]
@ganigeorgiev 

Could a similar approach be applied to API Filter rules? For example, I am trying to list records from "interests" collections for a user, but I want to exclude records that are already in user's "interests" relation.

Neither of these worked, `id != @request.auth.interests.id` or `id !~ @request.auth.interests.id`. However it works if I wanted to only list "interests" in a user's record (`id = @request.auth.interests.id`). Seems like I am missing something obvious.

**Reply by ganigeorgiev** [reply]
@shubhendumadhukar This may not work if `interests` is a multi-relation field because it will return `true` on the first matching `interests.id` condition.

Multi-match filter expressions and other filter enhancements are [currently in progress](https://github.com/orgs/pocketbase/projects/2) and most likely will be available with v0.11.0 (in the next 1-2 weeks if nothing critical shows up). _You can watch the [`filter-enhancements`](https://github.com/pocketbase/pocketbase/tree/filter-enhancements) branch._

**Reply by ganigeorgiev** [reply]
@shubhendumadhukar As a temporary workaround you can try a "not-contains" expression by checking the raw `interests` field value, aka.:

```
// notice that there is no `.id` suffix
id !~ @request.auth.interests

// or the opposite
@request.auth.interests !~ id
```

**Reply by shubhendumadhukar** [reply]
Thanks @ganigeorgiev. Of the provided workarounds, second one worked! (`@request.auth.interests !~ id`)

**Comment by shynome** [comment]
I have a think this filter `tags:each ?= "a" && tags:each ?= "c"` `tags:each` is the same sql var
if change it to `(tags:each ?= "a") && (tags:each ?= "c")` let this sql get 2 `tags:each` sql var, it will work

(sorry for my bad english

**Reply by ganigeorgiev** [reply]
@shynome You are commenting to a very old issue and I don't exactly understand what actually is the problem you are trying to explain.

`tags:each ?= "a" && tags:each ?= "c"` (no matter whether the expressions are in parenthesis) would technically always resolve to false because `tags:each` in both references represents a single "tags" value and it cannot be 2 things at the same time. It is a slightly different use case but see for example https://github.com/pocketbase/pocketbase/discussions/4319#discussioncomment-8420373/

**Reply by shynome** [reply]
I want select which `tags = ["a","c"]`, but today I have to use `tags ~ '"a"' && tags ~ '"c"'` to do it

**Reply by shynome** [reply]
Introduce variable scoping for parentheses `( )`, allowing each `tags:each` inside `(tags:each ?= "a") && (tags:each ?= "c")` to have its own distinct SQL variable name.
